### ISSUE TITLE
Remote Play Together don't work with Stem Play (Proton powered games)

### ISSUE THREAD

**Issue by NTMan** [ValveSoftware/steam-for-linux#6716]
#### Your system information

* Steam client version (build number or date): 1574216296
* Distribution (e.g. Ubuntu): Fedora
* Opted into Steam client beta?: Yes
* Have you checked for system updates?: Yes

#### Please describe your issue in as much detail as possible:
I bought and checked almost all the games presented on the Steam Remote Play Together sale.

| Game name | Issues |
| ----------------------| -------------------- |
| Guacamelee! Gold Edition | - |
| Gang Beasts | - |
| Lovers in a Dangerous Spacetime | - |
| STARWHAL | - |
| Super Cane Magic ZERO - Legend of the Cane Cane | - |
| ROCKETSROCKETSROCKETS | - |
| DERU - The Art of Cooperation | The game crashes when another player join by Remote Play Together |
| Push Me Pull You | Not working control at local client |
| Lichtspeer: Double Speer Edition | The game crashes when another player join by Remote Play Together |
| Riverbond (steam play) | Streaming not started |
| Chariot (steam play) | Streaming not started |
| Heave Ho (steam play) | Streaming not started |
| Catastronauts (steam play) | Streaming not started |
| Unruly Heroes (steam play) | Streaming not started |
| Degrees of Separation (steam play) | Streaming not started |
| The Adventure Pals (steam play) | Streaming not started |
| Rayman Legends (steam play) | Streaming only started under X session (Wayland not works) but gamepad not working at remote client |

How to interpret my results:
1) Streaming working even under Wayland if game is Linux native.
2) Streaming not working even under X session if the game Proton powered (exception of Rayman Legends but the game still has issues with controls)
3) Some native games crashes when another player joined to the game by Remote Play Together technology.

**Comment by dino-** [comment]
I've seen it change from allowing to disallowing any/all of my Windows games to stream from one Linux client to another. I don't think it has anything to do with client software updates either, I mean I've seen this change without the clients updating.

**Comment by rkost** [comment]
I had similar issues with games that require Proton to run. Video Stream does not start (although Audio is working). In my case: *LEGO MARVEL's Avengers*.

- Arch linux `5.3.6-1`
- Non Beta Steam Client (`1576550254`)
- Windows 10 as "guest" for remote play
- Proton Versions `4.2-9` and `4.11-11`

Fix for this particular game: Set DirectX version to 9 instead of 11 (in game option).

**Comment by jape42** [comment]
fallout 4 doesn't work with kubuntu host and client.  Proton 4.11-11.  Initial launch dialog is displayed zoomed in; once you click on "play" the video freezes but the audio continues.  The game seems to be working correctly on the host and can be interacted with using VNC.  Trying to connect with client after manually launching on host also results in video stream freezing.

**Comment by howels** [comment]
Tried this the other day and I have the same issue; audio stream is heard but video stream freezes on client after only a few frames.

**Comment by rkost** [comment]
> Tried this the other day and I have the same issue; audio stream is heard but video stream freezes on client after only a few frames.

Did you try selecting DirectX9 instead of 11?

**Comment by dgmvecuador** [comment]
Same issue, try "Granblue Fantasy: Versus" with Linux as host, it don't work friend can hear the sound but no image and no control detected; if the host is Windows, Linux client work properly .

**Comment by enricotagliavini** [comment]
One possible cause: using Intel iGPU by accident. This happened to me with Ticket to Ride (the Linux version is abandoned, so you have to use the Windows version with proton, which works very well with PROTON_NO_D3D11=1). It's not a graphic intensive game and I started it on a laptop with Intel iGPU + Nvidia card (optimus system). Client could barely hear the sound of the game and got a black screen.

Then I realized Steam Play needs the graphic card acceleration to do high performance real time encoding of the video to stream to the client, so I started the game with the Nvidia card instead via PRIME. It worked fine. On my desktop with a Radeon RX580 (no iGPU on this system) I can use Steam Play together with proton titles. Mostly tested with Carcassonne for now.

Bottom line: check that you have a graphic card capable of H.264 accelerated encoding and that it's working.

**Comment by roothorick** [comment]
I'm almost positive it's usually the Steam client failing to hook Vulkan, as in the vast majority of cases `PROTON_USE_WINED3D11=1" allows remote play to work. This isn't really a generally acceptable workaround because wined3d has compatibility and particularly performance problems.

**Comment by fallenguru** [comment]
> check that you have a graphic card capable of H.264 accelerated encoding and that it's working.

Software encoding works just fine. Get that running properly first, then you can try to get hardware encoding to work.

**Comment by dgmvecuador** [comment]
> I'm almost positive it's usually the Steam client failing to hook Vulkan, as in the vast majority of cases `PROTON_USE_WINED3D11=1` allows remote play to work. This isn't really a generally acceptable workaround because wined3d has compatibility and particularly performance problems.

Your are right, maybe vulkan is the problem, steam cannot capture vulkan for some reason

**Comment by romulasry** [comment]
Update on vulkan?

**Comment by romulasry** [comment]
Did PROTON_USE_WINED3D11=1 with Adventure Pals and it doesn't work.

**Comment by gitsmol** [comment]
This problem still exists. I'm quite keen to fix this and would appreciate any pointers for debugging. That way maybe I can at least find out what the problem is?

**Comment by jcnils** [comment]
I have the same problem with games under proton. Manjaro(Gnome)

The first time my friends try to connect, the game hangs [force quit / wait]. If I invite them again they can join.

They can:

- hear the audio (every time)
- control the game (often)

They cannot:

- see the game. (every time)


PROTON_USE_WINED3D11=1 doesn't work.

Please let me know if I can provide any more information to help debugging.

**Comment by SSUPII** [comment]
I confirm this still happening with Steam Deck host. The only games I can stream have native Linux builds, and all games I tested that run under Proton just remain on a fixed "Please Stand By" but can already be controlled.

`PROTON_USE_WINED3D11=1` makes all games tested crash immediately.
